dev/json_dto/pub.hpp

```diff
@@ -3019,6 +3019,53 @@ optional_no_default(
 			std::move( validator ) };
 }
 
+//! Create bind for an optional array mapping to fields, without default value.
+template < typename... types >
+class optional_array
+{
+	std::tuple<types&...> fields;
+
+public :
+	optional_array( types&... fields ) : fields{ fields... } {}
+
+	void read_from( const rapidjson::Value& array ) const
+	{
+		if ( array.IsNull() )
+			return;
+
+		if ( !array.IsArray() )
+			throw json_dto::ex_t{ "parent json type must be array" };
+
+		if ( array.Size() != sizeof...( types ) )
+			throw json_dto::ex_t{ "incorrect number of elements in array" };
+
+		auto read = [ this, &array ]<std::size_t... i>( std::index_sequence<i...> ) {
```

Comment:
Does it really works with C++14? It seems to be C++17 or even C++20 construct.

Reply:
Right, that won't work in c++14. 

It could probably be made to work if we unwrap the parameters using a construct like this:

```
template <std::size_t i>
void read(const rapidjson::Value& array)
{
    json_dto::read_json_value( std::get< i >( fields ), array[ i ] );
}

template <std::size_t i, typename... remaining>
void read(const rapidjson::Value& array)
{
    json_dto::read_json_value( std::get< i >( fields ), array[ i ] );
    read<remaining...>(array);
}

template <std::size_t... i>
void read(const rapidjson::Value& array, std::index_sequence<i...>)
{
    read<i...>(array);
}
```

But supporting older versions of c++ gets hairy quite fast!